python/app/shotgun_formatter.py

```diff
@@ -449,9 +449,16 @@ def get_link_filters(self, sg_location):
         if sg_location.entity_type in ["HumanUser"]:
             # the logic for users is different
             # here we want give an overview of their work
-            # for the current project 
-            link_filters.append(["project", "is", self._app.context.project])
-            
+            # for the current project
+
+            context_project = self._app.context.project
+
+            # When the current project is None, the user is in site context and
+            # we want to get the requested fields for all user's projects.
+            filter_operator = "is" if context_project else "is_not"
+
+            link_filters.append(["project", filter_operator, context_project])
```

Comment:
This would exclude things (tasks, versions, notes) that don't have a project associated. When you are in site mode, we want to show everything, so instead basically drop the project constraint completely.